validator/client/validator.go

```diff
@@ -330,73 +331,81 @@ func (v *validator) UpdateDuties(ctx context.Context, slot uint64) error {
 
 	v.duties = resp
 	v.logDuties(slot, v.duties.Duties)
-	subscribeSlots := make([]uint64, 0, len(validatingKeys))
-	subscribeCommitteeIDs := make([]uint64, 0, len(validatingKeys))
-	subscribeIsAggregator := make([]bool, 0, len(validatingKeys))
-	alreadySubscribed := make(map[[64]byte]bool)
-
-	for _, duty := range v.duties.Duties {
-		if duty.Status == ethpb.ValidatorStatus_ACTIVE || duty.Status == ethpb.ValidatorStatus_EXITING {
-			attesterSlot := duty.AttesterSlot
-			committeeIndex := duty.CommitteeIndex
+	go func() {
+		// Create a new context as the existing one will be canceled when the parent function exits.
```

Comment:
Can you break all of this out into another method?

Reply:
Can do; if I break it out simply the function call will look a little messy, _i.e._ `v.subscribeToSubnets(parentCtx, slot, validatingKeys, req)`  I was planning on a proper rewrite of this code to functions for the next part (with the removal of logging to a separate call there are two pieces of code that are virtually identical).  Three options:

  1. leave the code inline; deal with it in the next PR
  2. use the function call as outlined above; tidy up in the next PR
  3. bring forward the changes for the second phase in to this PR

Happy with any of these; let me know which you prefer.

Reply:
@prestonvanloon any thoughts on the above?